vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go

```diff
@@ -444,23 +422,12 @@ func resourceVcdNsxtIpSecVpnTunnelRead(_ context.Context, d *schema.ResourceData
 func resourceVcdNsxtIpSecVpnTunnelDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	vcdClient := meta.(*VCDClient)
 
-	// Handling locks is conditional. There are two scenarios:
-	// * When the parent Edge Gateway is in a VDC - a lock on parent Edge Gateway must be acquired
-	// * When the parent Edge Gateway is in a VDC Group - a lock on parent VDC Group must be acquired
-	// To find out parent lock object, Edge Gateway must be looked up and its OwnerRef must be checked
-	// Note. It is not safe to do multiple locks in the same resource as it can result in a deadlock
-	parentEdgeGatewayOwnerId, _, err := getParentEdgeGatewayOwnerId(vcdClient, d)
+	unlock, err := vcdClient.lockParentVdcGroupOrEdgeGateway(d)
 	if err != nil {
-		return diag.Errorf("[nsx-t ipsec vpn tunnel create/update] error finding parent Edge Gateway: %s", err)
+		return diag.Errorf("[nsx-t ipsec vpn tunnel create/update] %s", err)
```

Comment:
Looks to be a `delete` function, not "create/update"

Reply:
thanks for noticing, fixed.